### PR TITLE
WIP Tutorial Part Two: make CSS for Glamor example match others

### PR DESCRIPTION
**Pull request by hsribei**
The Tutorial Part Two example for Glamor is missing the `:last-child { marginBottom: 0 }` style that is present in both the CSS Modules and the Styled Components examples.

I added it and reformatted the line because it goes over 80 chars.

I need a little help with the rendering though. For some reason, this:

```javascript
  <div
    css={{
      display: `flex`,
      alignItems: `center`,
      margin: `0 auto 12px auto`,
      "&:last-child": { marginBottom: 0 }
    }}
  >
```

is getting turned into this:

gatsby js tutorial part two - gatsbyjs 2017-09-14 14-50-25

Further along, a mysterious newline also popped up that shouldn't be there (I didn't even change anything in that area):

gatsby js tutorial part two - gatsbyjs 2017-09-14 15-07-02 2

Is there anything I'm doing wrong?

### CODE REVIEW

**Comment by hsribei** [comment]
A deploy preview for the `www/` folder would be perfect.

**Comment by KyleAMathews** [comment]
Here you go :-) https://deploy-preview-2118--gatsbyjs.netlify.com/

It does get built (along with all the other example sites) just isn't being added here as a comment. Sooner than later I'll be revamping the example site building so they'll all show up in a single comment… (keep saying that…)

**Comment by KyleAMathews** [comment]
And thanks!

And on the formatting… there's something goofy about the line-highlighting css. I haven't had a chance to look at it. @fk if you have a sec sometime you might have an easier time figuring out where the extra lines are coming from…

**Comment by hsribei** [comment]
>It does get built (along with all the other example sites) just isn't being added here as a comment. 

That's good to know, thanks for the link.

And thanks for merging :) Hope to have contributions of more substance in the future.

**Comment by KyleAMathews** [comment]
Every bit counts! :-)

**Comment by fk** [comment]
@KyleAMathews @hsribei 👍

**Comment by fk** [comment]
@hsribei Forgot to say thanks! 🤗  You're def. not doing anything wrong! ;) When writing

> Further along, a mysterious newline also popped up that shouldn't be there (I didn't even change anything in that area):

Are you sure this was working correctly at some point?

@KyleAMathews Had only a few minutes, but looks like we somehow end up having nested `.gatsby-highlight-code-line` where the problem occurs, e.g. the "Bob Smith" User component:

image

I'd spontaneously say "seems a `<span>` in the "[code split](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-remark-prismjs/src/highlight-code.js#L21)" we're trying to highlight (wrap in a `span.gatsby-highlight-code-line`) isn't closed?" but that'd be 😩 .
Only looked at one console.log though.

**Comment by fk** [comment]
Yeah :( works just fine without our highlight wrapper `<span>` and that parent `<span>` has nested linebreaks:

image

image

**Comment by hsribei** [comment]
@fk you're welcome :)

>Are you sure this was working correctly at some point?

You're right, I just tested the site without the patch and the rogue newline before `</Container>` was already there.

**Comment by fk** [comment]
@hsribei Was fearing this, but only makes sense … :-(

@KyleAMathews So I took another look, ignoring the nagging thought that our approach to line-highlighting just won't work …

Adding CSS to target nested `.gatsby-highlight-code-line`s and adjusting the highlighted lines to cover the whole code block that the nesting occurs in got things working:

```jsx
      ".gatsby-highlight-code-line > .gatsby-highlight-code-line": {
        background: `red`,
        paddingRight: 0,
        marginRight: 0,
        display: `inline`,
      },
```

image

However if I choose to not highlight the last line of `<User />` (containing `/>`), we of course run into problems again:

image

TL;DR: Not a solution. As far as I can see for now, our approach – no matter if with or without the adjustments to make the highlighted line's background cover the whole width of the line – just won't work as long as we can't expect each line's tags (`<span>`s) to be closed in that same line.
